Mono.Addins/Mono.Addins/AddinEngine.cs

```diff
@@ -626,8 +626,8 @@ void RegisterAssemblyResolvePaths (RuntimeAddin addin, ModuleDescription descrip
 		{
 			lock (LocalLock) {
 				foreach (var asm in description.AssemblyNames) {
-					Debug.Assert(assemblyResolvePaths[asm] == addin);
-					assemblyResolvePaths[asm] = addin;
+					//Debug.Assert(assemblyResolvePaths[asm] == addin); This does not look right. Not called in old project since DEBUG symbol is not defined.
```

Comment:
I am not sure anymore, but I think the assert was supposed to pick up scenarios where there were multiple registrations of the same assembly.